### PR TITLE
Fix redundant assert

### PR DESCRIPTION
**Pull request by OlesiaSub**

### CODE REVIEW

File: plugin/src/main/kotlin/org/jetbrains/research/pynose/plugin/inspections/common/RedundantAssertionTestSmellVisitor.kt [hunk 1]
```diff
@@ -32,22 +35,35 @@ open class RedundantAssertionTestSmellVisitor(holder: ProblemsHolder?, session:
...
-            return
+        if (args[0] is PyBinaryExpression) {
+            binaryExpression = args[0] as PyBinaryExpression
         }
-        val binaryExpression = args[0] as PyBinaryExpression
-        val psiOperator = binaryExpression.psiOperator ?: return
-        if (binaryExpression.children.size < 2) {
-            return
+        if (args[0] is PyParenthesizedExpression) {
+            literalExpression = PsiTreeUtil.findChildOfType(args[0], PyLiteralExpression::class.java)
```
Comment (SmirnovOleg) [inline plugin/src/main/kotlin/org/jetbrains/research/pynose/plugin/inspections/common/RedundantAssertionTestSmellVisitor.kt:47]: Check nested parentheses?

**Comment by OlesiaSub** [comment]
@SmirnovOleg I added parentheses processing and supported assertions like self.assertTrue(4 < 5)
I assumed that any unittest assertion with one parameter (self.assertXXX) should be highlighted in case it is called on a literal expression (like self.assertTrue(5)). And as for assertions with 2 args: if both args are literal expressions - the assertion will be highlighted (i.e. self.assertEqual(2, "hello")).

File: example/tests/test_something.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def test_something():
```
Comment (SmirnovOleg) [inline example/tests/test_something.py:1]: Nice name btw

File: plugin/src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -185,7 +185,9 @@
                          displayName="Test maverick"
                          shortName="TestMaverickTestSmellPytestInspection"/>
     </extensions>
-
+    <extensions defaultExtensionNs="com.intellij">
+        <appStarter implementation="org.jetbrains.research.pynose.headless.HeadlessRunner"/>
```
Comment (SmirnovOleg) [inline plugin/src/main/resources/META-INF/plugin.xml:189]: We don't have in `master`, do we? Why is it here O_o?

**Review by SmirnovOleg — CHANGES_REQUESTED** [review]

File: example/tests/test_something.py (hunk 2 above)
Comment (OlesiaSub) [reply example/tests/test_something.py:1]: thanks?

umm is it a problem if almost all of the names in my tests look like SomeClass test_something test_something_else etc?)))

File: plugin/src/main/resources/META-INF/plugin.xml (hunk 3 above)
Comment (OlesiaSub) [reply plugin/src/main/resources/META-INF/plugin.xml:189]: we do have it in master :')
we only removed ignored, unknown and general fixture inspections

Comment (OlesiaSub) [reply plugin/src/main/resources/META-INF/plugin.xml:189]: oh wait no we don't, sorry

File: example/tests/test_something.py (hunk 2 above)
Comment (SmirnovOleg) [reply example/tests/test_something.py:1]: :)
It's not very good, but it's not terrible though. Let's leave it as is.

**Pull request merged**